src/clandinin_lab_to_nwb/brezovec/brezovecimagingextractor.py

```diff
@@ -58,118 +60,123 @@ def get_streams(cls, folder_path: PathType) -> dict:
-            )
+        self.folder_path = Path(folder_path)
 
-            file_names =list(folder_path.glob("*.nii")) 
-            # can't extract the names from xml --> they are listed as .ome.tif 
-            #TODO: to be implemented in a way that is not dependent to the filename
+        self._xml_root = _parse_xml(folder_path=folder_path)
 
-            channel_id = streams["channel_streams"][stream_name]
-            file_for_stream = [file.as_posix() for file in file_names if 'channel_'+channel_id in file.as_posix()]
-            self._nii = self._niftifile.load(file_for_stream[0])
+        # TODO: All the checks on the channel_names, streams
+        self.stream_name = stream_name
+        streams = self.get_streams(folder_path=folder_path)
+        self._channel_names = list(streams["channel_streams"].keys())
+        assert (
+            stream_name in self._channel_names
+        ), f"The selected stream '{stream_name}' is not in the available channel stream '{self._channel_names}'!"
 
-            self._num_rows = self._nii.shape[0]  
-            self._num_columns = self._nii.shape[1]
-            self._num_planes_per_channel_stream = self._nii.shape[2]
+        file_names = list(folder_path.glob("*.nii"))
+        # can't extract the names from xml --> they are listed as .ome.tif
+        # TODO: to be implemented in a way that is not dependent to the filename
 
-            self._num_frames = self._nii.shape[-1]
+        channel_id = streams["channel_streams"][stream_name]
+        file_for_stream = [file.as_posix() for file in file_names if "channel_" + channel_id in file.as_posix()]
+        self.nibabel_image = self._niftifile.load(file_for_stream[0])
 
-            sampling_frequency = self.get_sampling_frequency()
-            assert sampling_frequency is not None, "Could not determine the frame rate from the XML file."
-            self._sampling_frequency = sampling_frequency
+        self._num_rows = self.nibabel_image.shape[0]
+        self._num_columns = self.nibabel_image.shape[1]
+        self._num_planes_per_channel_stream = self.nibabel_image.shape[2]
 
+        self._num_frames = self.nibabel_image.shape[-1]
 
-            self.xml_metadata = self._get_xml_metadata()
+        sampling_frequency = self.get_sampling_frequency()
+        assert sampling_frequency is not None, "Could not determine the frame rate from the XML file."
+        self._sampling_frequency = sampling_frequency
 
+        self._times = None
+        self.xml_metadata = self._get_xml_metadata()
 
     def get_image_size(self) -> Tuple[int, int, int]:
-        return (self._num_rows, self._num_columns,self._num_planes_per_channel_stream)
-    
+        return (self._num_rows, self._num_columns, self._num_planes_per_channel_stream)
+
     def get_num_frames(self) -> int:
         return self._num_frames
 
     def get_dtype(self) -> DtypeType:
-        return self._nii.get_data_dtype()
+        return self.nibabel_image.get_data_dtype()
```

Comment:
You had this variable with _ in the init but then without the _ in the get_video. 

I also changed the name to a more explicit one so this error is not hidden in a three letter variable.